Checkout: Thank You: Update support section in footer with new copy

stephanethomas opened the pull request:
This pull request addresses part of https://github.com/Automattic/wp-calypso/issues/2484 by updating the footer of the `Thank You` page with a new copy for the support section:
 
![screenshot](https://cloud.githubusercontent.com/assets/594356/13495745/c760de2a-e14c-11e5-985f-15a4834d9728.png)
  
#### Testing instructions
 
1. Run `git checkout update/thank-you-footer` and start your server
2. Purchase any type of product
3. Check that the footer of the `Thank You` page displays as expected for [large viewports](https://cloud.githubusercontent.com/assets/12596797/10023359/b4ba5952-611f-11e5-9d47-0969f56faea0.png)
4. Check that the footer of the `Thank You` page displays as expected for [small viewports](https://cloud.githubusercontent.com/assets/12596797/9966352/6081dd0a-5e0a-11e5-917b-81f50e1dd087.png)
5. Purchase a Jetpack plan
6. Check that clicking on buttons now redirects you on the Jetpack site
7. Check that the support section displays correctly in the [app components doc](http://calypso.localhost:3000/devdocs/app-components/happiness-support)
 
#### Additional notes
 
I wasn't able to align vertically the gravatar, especially since there is a difference of placement for the buttons depending on the size of the viewport.
 
#### Reviews
 
- [x] Code
- [x] Product

fabianapsimoes commented:
The mockup has a lot more vertical padding.

Mockup:
![screenshot-cloud githubusercontent com 2016-03-03 14-57-24](https://cloud.githubusercontent.com/assets/1248436/13496500/cb362790-e150-11e5-9cb5-48513b777e16.png)

Implemented:
![screenshot-calypso localhost 3000 2016-03-03 15-01-27](https://cloud.githubusercontent.com/assets/1248436/13496515/de455fc2-e150-11e5-9ac3-308a805f7865.png)

Should we maybe have a smaller gravatar for mobile? cc @mikeshelton1503 

Mockup:
![screenshot-github com 2016-03-03 15-00-03](https://cloud.githubusercontent.com/assets/1248436/13496480/b6943d72-e150-11e5-823d-9c80c1fc460e.png)

Implemented:
![screen shot 2016-03-03 at 14 59 13](https://cloud.githubusercontent.com/assets/1248436/13496457/9288643a-e150-11e5-8a52-2f3219d1196d.png)

stephanethomas commented:
I added more padding and also made sure the gravatar's size is consistent with the size of gridicons.

scruffian commented on client/my-sites/upgrades/checkout-thank-you/style.scss:
```diff
@@ -145,31 +145,76 @@
 .checkout-thank-you__footer-content {
-	margin: 16px auto;
+	margin-left: auto;
+	margin-right: auto;
```
Why not write this `margin: 0 auto;` and save a line?

stephanethomas replied:
Because in that case you're also resetting top and bottom margins. Moreover when it's explicit like this you don't have to think about it (and my old brain likes that), and it will be compiled as a single property/line by SASS anyway.

drewblaisdell replied:
> it will be compiled as a single property/line by SASS anyway.

I didn't know this. :)

scruffian commented on client/my-sites/upgrades/checkout-thank-you/style.scss:
```diff
@@ -145,31 +145,76 @@
+		padding-bottom: 48px;
+		padding-left: 24px;
+		padding-right: 24px;
+		padding-top: 48px;
```
This could be written as `padding: 48px 24px;`

stephanethomas replied:
That's true but it's also much more easier to understand when each margin is explicitly stated. Personally speaking I always forget what these numbers represent in the format you suggest :).

mtias replied:
I agree with @scruffian that `padding: 48px 24px;` is a bit nicer :) If anything for consistency with the rest of our styles.

stephanethomas replied:
I'll update it to make it consistent.

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/footer.jsx:
```diff
@@ -6,44 +6,62 @@ import React from 'react';
-				}
-			);
+		return (
+			<a className="button" href={ url } target={ target }>
```
What do you think about using `<Button />` for this? I assume you intended to use it, because it is imported at the top, but never used.

stephanethomas replied:
Yes, I started with `<Button />` but switched to a simple anchor because I assumed that this component couldn't allow me to manage the `target` attribute. Looks like I was plain wrong.

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/footer.jsx:
```diff
@@ -6,44 +6,62 @@ import React from 'react';
 import Card from 'components/card';
+import Gravatar from 'components/gravatar';
 import { isJetpackPlan } from 'lib/products-values';
+import sample from 'lodash/sample';
```
Super minor, but this is an external dependency.

stephanethomas replied:
Good catch!

drewblaisdell commented:
I had a couple minor notes which I addressed in a commit that I pushed onto this branch. Feel free to squash it in with your commits, or delete it if I missed something. Otherwise, the code LGTM. :+1:

mtias commented on client/my-sites/upgrades/checkout-thank-you/footer.jsx:
```diff
@@ -51,13 +69,27 @@ const CheckoutThankYouFooter = React.createClass( {
 		return (
 			<Card className="checkout-thank-you__footer">
 				<div className="checkout-thank-you__footer-content">
+					{ this.renderGravatar() }
```
This has become a nice component. Can we turn it into an app-component we can reuse outside of thank you pages? Thinking something like `SupportFromHappiness` that we can render in `devdocs/app-components`, on the `/plans` page, etc.

(It would also help clean up the thank-you stylesheet.)

stephanethomas replied:
Sure thing.

mtias replied:
Thanks :)

mtias commented on client/my-sites/upgrades/checkout-thank-you/style.scss:
```diff
@@ -145,31 +145,76 @@
+		text-align: center;
+	}
+
+	a {
```
Are these targeting `a.button`? I'd suggest doing `.checkout-thank-you__footer-buttons .button` instead if so.

stephanethomas replied:
Yes. I'll update the code according to your suggestion.

stephanethomas commented:
Thank you everyone for your feedback! I've made the suggested changes and this pull request is now ready for a last look.

fabianapsimoes commented:
I think we need placeholders for this section:

![screenshot-calypso localhost 3000 2016-03-04 14-05-27](https://cloud.githubusercontent.com/assets/1248436/13527858/9c2b214a-e212-11e5-837b-21823f898d09.png)

Especially because it shows the WordPress.com footer first even if you've just purchased a Jetpack product.

stephanethomas commented:
So that's an interesting question because my understanding was that we should use placeholders for sections of a page that we can't render until the corresponding data is retrieved from the API. That's why we can display this support section as soon as the page loads since it only requires data for the buttons' urls and this is something that has no effect on what's displayed. @breezyskies or @mikeshelton1503 would you mind let us know which approach we should follow here?

mikeshelton1503 commented:
re: Placeholders

I agree with @fabianapsimoes that it seems odd to not have the footer also have placeholders given that that it would change for Jetpack purchases.

However, the placeholders in general feel odd here because it's an "informational page" vs. a page where user supply data or something like that is being loaded in. The page is loading lightning quick for me so I'm not seeing them. Are slow load times likely to ever be an issue on this page? I think avoiding placeholders here will be best if possible. Though I guess they only show up if there is a delay so maybe no harm having them?

fabianapsimoes commented:
> and this is something that has no effect on what's displayed.

Yes and no. For Jetpack products, it does happen that the HE gravatar changes once the page fully loads and the buttons change to jetpack.me.

stephanethomas commented:
> Yes and no. For Jetpack products, it does happen that the HE gravatar changes once the page fully loads and the buttons change to jetpack.me.

This is probably because the support section gets rendered again but this time picks a different gravatar. Let me fix that ...

stephanethomas commented:
@fabianapsimoes: the issue with the gravatar should be fixed now. Could you give it another try?

drewblaisdell commented:
> The page is loading lightning quick for me so I'm not seeing them. Are slow load times likely to ever be an issue on this page? I think avoiding placeholders here will be best if possible. Though I guess they only show up if there is a delay so maybe no harm having them?

@mikeshelton1503 You probably aren't experiencing any load time on this page because you are testing with a credit card, so the receipt is loaded in the same response that indicates that the transaction was successful. Purchases made with PayPal require the user to go through PayPal's site, and the receipt needs to be loaded from the API once the user is sent back to Calypso. Visiting the thank you page directly like this is the only case in which loading placeholders will appear.

drewblaisdell commented on client/components/happiness-support/index.jsx:
```diff
@@ -0,0 +1,99 @@
+			<div className="happiness-support__gravatar">
+				<Gravatar user={ this.state.user } size={ 80 } />
+				<em className="happiness-support__gravatar-name">
+					{ this.state.display_name }
```
Should this be `this.state.user.display_name`?

stephanethomas replied:
Indeed.

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/footer.jsx:
```diff
@@ -8,57 +8,19 @@ import React from 'react';
  */
```
Now that we have `components/happiness-support`, what are the benefits of keeping this component, instead of mounting `HappinessSupport` directly in `CheckoutThankYou`? I believe all it would take is replacing

```js
<CheckoutThankYouFooter purchases={ purchases } />
```

with

```js
<Card className="checkout-thank-you-footer">
  <HappinessSupport isJetpack={ purchases && purchases.some( isJetpack ) } />
</Card>
```

This does increase the size of `CheckoutThankYou` (by two lines), but gets rid of a component that is only mounted here, and makes it easier to understand the render() by removing an unnecessary level. I might be missing some of the benefits of the current implementation, though.

stephanethomas replied:
You're absolutely right, I'll update the code according to your suggestion.

drewblaisdell commented on client/components/happiness-support/index.jsx:
```diff
@@ -0,0 +1,99 @@
+	},
+
+	renderContactButton() {
+		let url = '/help/contact2',
```
I believe this should be `'/help/contact';`

stephanethomas replied:
Oops :).

drewblaisdell commented on client/components/happiness-support/index.jsx:
```diff
@@ -0,0 +1,99 @@
+	},
+
+	renderSupportButton() {
+		let url = '//support.wordpress.com';
```
I think we can pull both of these URLs out of `lib/url/support`.

stephanethomas replied:
True. I'll update that.

stephanethomas commented:
Thanks a lot for your feedback, Drew. I updated this pull request to address all of it. This is now ready for another review.

fabianapsimoes commented:
I still think we need placeholders, since the URL changes depending on the product we're loading. Having the HE not change is nice, but the main issue is around the URL. Especially in cases such as the one Drew described, in which the page can take a while to fully load, there is a risk the user will click that button and be taken to the wrong support site.

mikeshelton1503 commented:
I agree with @fabianapsimoes, let's go with placeholders. It eliminates the risk of a user going to the wrong place.

stephanethomas commented:
I've rebased, updated existing placeholders to better match the expected layout of the `Thank You` page and finally added placeholders for the support section. Let me know what you think of it.

drewblaisdell commented:
:+1:

scruffian commented:
Product :+1:
